FOUR-2687

GooseCoder opened the pull request:
Fixing missing config object error.

sanjacornelius commented:
Hey @GooseCoder! Two more observations.

1. When adding an Upload File control I am now getting this error in the console. I switched to the `develop` branch to see if this error was already occurring, but it only happens within this branch.

```
vue.esm.js?a026:628 [Vue warn]: Unknown custom element: <RequiredCheckbox> - did you register the component correctly? For recursive components, make sure to provide the "name" option.

found in

---> <VueFormBuilder> at src/components/vue-form-builder.vue
       <App> at src/App.vue
         <Root>
```

2. The same message that is displayed when viewing preview mode also needs to be displayed in edit mode. The Upload File component should only appear when the screen is running within a process/request. 

https://www.loom.com/share/8bdc7ec37d1c4c088dda4b425d128e8a

GooseCoder commented:
> Hey @GooseCoder! Two more observations.
> 
>     1. When adding an Upload File control I am now getting this error in the console. I switched to the `develop` branch to see if this error was already occurring, but it only happens within this branch.
> 
> 
> ```
> vue.esm.js?a026:628 [Vue warn]: Unknown custom element: <RequiredCheckbox> - did you register the component correctly? For recursive components, make sure to provide the "name" option.
> 
> found in
> 
> ---> <VueFormBuilder> at src/components/vue-form-builder.vue
>        <App> at src/App.vue
>          <Root>
> ```
> 
>     1. The same message that is displayed when viewing preview mode also needs to be displayed in edit mode. The Upload File component should only appear when the screen is running within a process/request.
> 
> 
> **Video**
> https://www.loom.com/share/8bdc7ec37d1c4c088dda4b425d128e8a

Hi @sanjacornelius  
Following David advice. I've added a copy of the RequiredCheckbox from the core repo and integrated into this branch. The error should not be there anymore while testing on the standalone screen-builder app.

sanjacornelius commented:
@GooseCoder Looks good and works as expected!
